tests/unit/package_managers/test_pip.py

```diff
@@ -58,116 +54,116 @@ def make_dpi(
     )
 
 
-@pytest.mark.parametrize("toml_exists", [True, False])
-@pytest.mark.parametrize("toml_name", ["name_in_pyproject_toml", None])
-@pytest.mark.parametrize("toml_version", ["version_in_pyproject_toml", None])
-@pytest.mark.parametrize("py_exists", [True, False])
-@pytest.mark.parametrize("py_name", ["name_in_setup_py", None])
-@pytest.mark.parametrize("py_version", ["version_in_setup_py", None])
-@pytest.mark.parametrize("cfg_exists", [True, False])
-@pytest.mark.parametrize("cfg_name", ["name_in_setup_cfg", None])
-@pytest.mark.parametrize("cfg_version", ["version_in_setup_cfg", None])
-@pytest.mark.parametrize("repo_name_with_subpath", ["bar-subpath", None])
-@mock.patch("cachi2.core.package_managers.pip.SetupCFG")
-@mock.patch("cachi2.core.package_managers.pip.SetupPY")
+@pytest.mark.parametrize(
+    "dynamic_version",
+    [True, False],
+)
 @mock.patch("cachi2.core.package_managers.pip.PyProjectTOML")
-@mock.patch("cachi2.core.package_managers.pip.get_repo_id")
-def test_get_pip_metadata(
-    mock_get_repo_id: mock.Mock,
+def test_get_pip_metadata_from_pyproject_toml(
```

Comment:
You decreased the value of the unit test by adding per-project file test case because you never test the combinations by making the main test split into multiple ones which is surely more readable but only part of the original logic is being tested, that was the point of the complex test.